presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -309,13 +313,49 @@ public Node visitQuerySpecification(SqlBaseParser.QuerySpecificationContext cont
                 new Select(isDistinct(context.setQuantifier()), visit(context.selectItem(), SelectItem.class)),
                 from,
                 visitIfPresent(context.where, Expression.class),
-                visit(context.groupBy, Expression.class),
+                visit(context.groupingElement(), GroupBySpecification.class),
                 visitIfPresent(context.having, Expression.class),
                 ImmutableList.of(),
                 Optional.<String>empty());
     }
 
     @Override
+    public Node visitGroupingColumnReferenceList(SqlBaseParser.GroupingColumnReferenceListContext context)
+    {
+        return new GroupingColumnReferenceList(context.qualifiedName().stream().map(AstBuilder::getQualifiedName).collect(Collectors.toList()));
+    }
+
+    @Override
+    public Node visitSimpleGroupByExpressions(SqlBaseParser.SimpleGroupByExpressionsContext context)
+    {
+        return new SimpleGroupBy(visit(context.expression(), Expression.class));
+    }
+
+    @Override
+    public Node visitRollupList(SqlBaseParser.RollupListContext context)
+    {
+        throw new IllegalArgumentException("ROLLUP is not yet supported");
```

Comment:
Since you have introduced the class `Rollup`, this should be implemented? ditto for the following ones.